packages/devextreme/js/__internal/ui/counter/counter.component.tsx

```diff
@@ -0,0 +1,81 @@
+// eslint-disable-next-line max-classes-per-file
+import { BaseInfernoComponent } from '@ts/core/r1/runtime/inferno';
+import { Widget } from '@ts/core/r1/widget';
+import withMethods from '@ts/ui/counter/counter.methods_hoc';
+import { CounterMethodsProvider } from '@ts/ui/counter/counter.methods_provider';
+import { CounterStepProvider } from '@ts/ui/counter/counter.methods_provider2';
+import type { CounterProps } from '@ts/ui/counter/counter.types';
```

Comment:
```suggestion
import type { CounterProps } from '@ts/ui/counter/types';
```

in my opinion duplicating namespace in folder name and in file name is bad